pkg/client/supplier/client.go

```diff
@@ -59,79 +58,91 @@ func NewSupplierClient(
 	return sClient, nil
 }
 
-// SubmitProof constructs a submit proof message then signs and broadcasts it
-// to the network via #txClient. It blocks until the transaction is included in
-// a block or times out.
-func (sClient *supplierClient) SubmitProof(
+// SubmitProofs constructs submit proof messages into a single transaction
+// then signs and broadcasts it to the network via #txClient. It blocks until
+// the transaction is included in a block or times out.
+func (sClient *supplierClient) SubmitProofs(
 	ctx context.Context,
-	sessionHeader sessiontypes.SessionHeader,
-	proof *smt.SparseMerkleClosestProof,
+	sessionProofs []*relayer.SessionProof,
 ) error {
 	logger := polylog.Ctx(ctx)
 
-	proofBz, err := proof.Marshal()
-	if err != nil {
-		return err
-	}
+	msgs := make([]cosmostypes.Msg, len(sessionProofs))
 
-	// TODO(@bryanchriswhite): reconcile splitting of supplier & proof modules
-	//  with off-chain pkgs/nomenclature.
-	msg := &prooftypes.MsgSubmitProof{
-		SupplierAddress: sClient.signingKeyAddr.String(),
-		SessionHeader:   &sessionHeader,
-		Proof:           proofBz,
+	for i, sessionProof := range sessionProofs {
+		// TODO(@bryanchriswhite): reconcile splitting of supplier & proof modules
+		//  with off-chain pkgs/nomenclature.
+		msg := &prooftypes.MsgSubmitProof{
+			SupplierAddress: sClient.signingKeyAddr.String(),
+			SessionHeader:   sessionProof.SessionHeader,
+			Proof:           sessionProof.ProofBz,
+		}
+
+		msgs[i] = msg
 	}
-	eitherErr := sClient.txClient.SignAndBroadcast(ctx, msg)
+
+	eitherErr := sClient.txClient.SignAndBroadcast(ctx, msgs...)
 	err, errCh := eitherErr.SyncOrAsyncError()
 	if err != nil {
 		return err
 	}
 
-	// TODO_IMPROVE: log details related to what & how much is being proven
-	logger.Info().
-		Fields(map[string]any{
-			"supplier_addr": sClient.signingKeyAddr.String(),
-			"app_addr":      sessionHeader.ApplicationAddress,
-			"session_id":    sessionHeader.SessionId,
-			"service":       sessionHeader.Service.Id,
-		}).
-		Msg("submitted a new proof")
+	for _, sessionProof := range sessionProofs {
+		sessionHeader := sessionProof.SessionHeader
+		// TODO_IMPROVE: log details related to what & how much is being proven
+		logger.Info().
+			Fields(map[string]any{
+				"supplier_addr": sClient.signingKeyAddr.String(),
+				"app_addr":      sessionHeader.ApplicationAddress,
+				"session_id":    sessionHeader.SessionId,
+				"service":       sessionHeader.Service.Id,
+			}).
+			Msg("submitted new proofs")
```

Comment:
This would be logged once per proof:

```suggestion
			Msg("submitted new proof")
```